src/esp/sensor/VisualSensor.cpp

```diff
@@ -22,5 +28,21 @@ void VisualSensor::bindRenderTarget(gfx::RenderTarget::uptr&& tgt) {
   tgt_ = std::move(tgt);
 }
 
+void VisualSensor::updateResolution(int height, int width) {
+  spec_->resolution = {height, width};
+}
+
+void VisualSensor::updateResolution(vec2i resolution) {
+  spec_->resolution = resolution;
+}
+
+bool operator==(const VisualSensorSpec& a, const VisualSensorSpec& b) {
+  return a.uuid == b.uuid && a.sensorType == b.sensorType && a.near == b.near &&
```

Comment:
you have the operator == function in base class as well. why not use it to check the variables defined the base class?